moped-editor/src/views/projects/projectView/ProjectComponents/mapUtils.js

```diff
@@ -0,0 +1,251 @@
+import { useMemo } from "react";
+import { Source, Layer } from "react-map-gl";
+import { basemaps, SOURCES } from "./mapSettings";
+import { MAP_STYLES } from "./mapStyleSettings";
+
+const mapStyles = MAP_STYLES;
+
+export const useComponentFeatureCollection = (component) =>
+  useMemo(() => {
+    if (!component || !component?.features) return null;
+    return { type: "FeatureCollection", features: component.features };
+  }, [component]);
+
+/**
+ * Iterate through the mapStyles config to create an array of interactive layers
+ */
+export const interactiveLayerIds = Object.entries(mapStyles).reduce(
+  (acc, [key, value]) => {
+    if (value.isInteractive) {
+      acc.push(key);
+    }
+
+    return acc;
+  },
+  []
+);
+
+/**
+ * Component that renders Mapbox source and layers needed for the aerial basemap
+ * @param {string} basemapKey
+ * @returns JSX.Element
+ */
+export const BaseMapSourceAndLayers = ({ basemapKey }) => {
+  return (
+    <>
+      <Source {...basemaps.aerial.sources.aerials} />
+      <Layer
+        {...basemaps.aerial.layers.aerials}
+        layout={{ visibility: basemapKey === "aerial" ? "visible" : "none" }}
+      />
+      {/* Always show street labels so they can be the "target" of beforeId 
+      and always appear on top of everything else */}
+      <Layer
+        {...{
+          ...basemaps.aerial.layers.streetLabels,
+          layout: {
+            ...basemaps.aerial.layers.streetLabels.layout,
+          },
+        }}
+      />
```

Comment:
I agree with John that it's much cleaner to have the labels in a component like this, but I am concerned about their legibility on the street layer. 

![image](https://user-images.githubusercontent.com/3653206/197620657-74a0fb04-894d-42c2-946b-c6ce8b6dd126.png)

This white-with-black-stroke is on point for aerial imagery,  but I find it hard to read on the much lighter street layer. 

If there's agreement, would it be something that could be addressed via conditional styling of the labels or maybe a seperate layer defined for them and used conditionally?

Reply:
@frankhereford I agree on this the more and more that I work on this. It makes my eyes feel unfocused 👓 so I'll create an issue to get the labels from the streets layer and show those conditionally. Thank you! 🙏